core/_interfaces/test-complete.i.ts

```diff
@@ -0,0 +1,14 @@
+import { ITestCase, ITest, ITestFixture } from "./index";
+import { TestOutcome } from "../results/test-outcome";
+
+export interface  ITestCompleteEvent {
+    currentTestIndex: number;
```

Comment:
Could we change this to `testId` as I think the word "current" may cause a bit of confusion in the scope of the asynchronous events and there is potential for the indexes to run out of sync with some asynchronous tests in the future so you may get for example 2, 3, 1 in that order.

Also could we remove `totalNumberOfTests` for now as the total may be different in different contexts and should be available to the caller so shouldn't need to give it back in the context of this event